### PR TITLE
Widgetpreviews

### PR DESCRIPTION
**Pull request by bobmitch**
Add filtering and previews for widget override method on page options.

### CODE REVIEW

File: admin/controllers/pages/views/edit/view.php [hunk 1]
```diff
@@ -85,7 +85,72 @@
...
+    gap: 1rem;
+    padding: 1rem;
+    box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
+	justify-content: space-between;
+	border-radius: 0.5rem;
+}
+.widget_actions {
+	/* display:flex;
+	gap:1rem; */
+}
```
Comment (workaj) [inline admin/controllers/pages/views/edit/view.php:110]: dead css that is duplicated below

File: admin/controllers/pages/views/edit/view.php [hunk 2]
```diff
@@ -345,6 +425,48 @@
 
 <script>
 
+	// preview widget
+	function preview_widget(el) {
+		fetch(<?php echo Config::uripath();?>'/admin/pages/edit/widget_preview/' + el.dataset.widgetid).then(function (response) {
+			return response.text();
+		}).then(function (html) {
```
Comment (workaj) [inline admin/controllers/pages/views/edit/view.php:432]: }).then((html)=>{

File: admin/controllers/pages/views/edit/view.php [hunk 3]
```diff
@@ -345,6 +425,48 @@
...
+			if (info_text.includes(search_value.toLowerCase()) || !search_value) {
+				// show
+				wrap_el.style.display="flex";
+			}
+			else {
+				wrap_el.style.display="none";
+			}
+		});
+	}
+	document.querySelector('#widget_title_filter')?.addEventListener('input', function(e){
```
Comment (workaj) [inline admin/controllers/pages/views/edit/view.php:466]: document.querySelector('#widget_title_filter')?.addEventListener('input', (e)=>{

**Review by workaj — CHANGES_REQUESTED** [review]
minor changes

**Comment by bobmitch** [comment]
Requested changes made in https://github.com/HoltBosse/Alba/pull/434/commits/9c98c6578afe09692d5b4cb5ff792b75515a0c85

File: admin/controllers/pages/views/edit/view.php [hunk 4]
```diff
@@ -345,6 +421,48 @@
 
 <script>
 
+	// preview widget
+	function preview_widget(el) {
+		fetch(<?php echo Config::uripath();?>'/admin/pages/edit/widget_preview/' + el.dataset.widgetid).then(function (response) {
```
Comment (workaj) [inline admin/controllers/pages/views/edit/view.php:426]: (response)=>{

Comment (bobmitch) [reply admin/controllers/pages/views/edit/view.php]: Change in https://github.com/HoltBosse/Alba/pull/434/commits/cb1854b5e8197b7abf23808e50fa8706c4874900

**Pull request merged**